### PR TITLE
Handle cases where we might fail silently

### PR DESCRIPTION
**Pull request by neonichu**
There are a few cases in which we are throwing `Diagnostics.fatalError` but aren't actually sure that there were previous diagnostics, so we could be failing silently. This is not only theoretical, I just ran into one such case.

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Sources/Build/BuildOperation.swift [hunk 1]
```diff
@@ -163,6 +168,7 @@ public final class BuildOperation: PackageStructureDelegate {
             ) {
                 return result
             }
+            diags.emit(error: "computing llbuild target name failed")
```
Comment (aciidb0mb3r) [inline Sources/Build/BuildOperation.swift]: Should probably wrap this in `!diags.hasErrors` as well? Also, maybe reword to "internal error: failed to compute the llbuild target name"
